src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,453 @@
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit_mask (int:bit value): 0 = output | 1 = input
+        """
+        if self.productID == 0x0075 and port_number == usb_1208LS.DIO_AUXPORT:
+            bit_mask = ((bit_mask ^ 0xff) & 0xff)
+
+        self.DOut(port_number, bit_mask)
+
+    def DIn(self, port_number):
+        """
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :return: the value seen at the port pins
+        """
+        DIO_port = self._return_port_config(port_number)
+        return sum(v << i for i, v in enumerate(DIO_port))
+
+    def DOut(self, port_number, value):
+        """
+        This command writes data to the DIO port bits that are configured as outputs.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param value: value to write to the port (0-255)
+        """
+        if value < 0 or value > 255:
+            raise ValueError("Value to set is not between 0 and 255")
+
+        DIO_port = self._return_port_config(port_number)
+        in_val = format(value, '#010b')
+        in_val = in_val.replace("0b", "")
+
+        for num, c in enumerate(in_val[::-1]):
+            DIO_port[num] = int(c)
+
+    def DBitIn(self, port_number, bit):
+        """
+        This command reads an individual digital port bit.  It will return the
+        value seen at the port pin, so may be used for an input or output bit.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit: the bit to read (0-7)
+        :return (int): value 0 or 1
+        """
+        if bit < 0 or bit > 7:
+            raise ValueError("Bit value is not between 0 and 7")
+
+        DIO_port = self._return_port_config(port_number)
+
+        return DIO_port[bit]
+
+    def DBitOut(self, port_number, bit, value):
+        """
+        This command writes an individual digital port bit.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit: the bit to read (0-7)
+        :param value: the value to write to the bit (0 or 1)
+        """
+        if bit < 0 or bit > 7:
+            raise ValueError("Bit value is not between 0 and 7")
+        if value < 0 or value > 1:
+            raise ValueError("Value to set should be either 0 or 1")
+
+        DIO_port = self._return_port_config(port_number)
+        DIO_port[bit] = value
+
+    def AOut(self, channel, value):
+        """
+        This command sets the voltage output of the specified analog output channel
+        :param channel: selects output channel (0 or 1)
+        :param value: value (uint16) in counts to output [10-bits 0-5V]
+        """
+        if channel > 1 or channel < 0:
+            channel = 0
+        # force automatic clipping
+        if (value > 0x3ff):
+            value = 0x3ff
+        if (value < 0):
+            value = 0
```

Comment:
Clipping can be done in 1 line:
```suggestion
        value  = min(max(0, value), 0x3ff)
```